pkg/commands/plan/plan_run.go

```diff
@@ -22,36 +22,214 @@ import (
+	set := cli.NewFlagSet(c.testFlagSetOpts...)
+
+	c.GitHubFlags.AddFlags(set)
+	c.RetryFlags.AddFlags(set)
+
+	f := set.NewSection("Command options")
+
+	f.StringVar(&cli.StringVar{
+		Name:    "working-directory",
+		Target:  &c.flagWorkingDirectory,
+		Example: "terraform",
+		Usage:   "The working directory for Guardian to execute commands in.",
+	})
+
+	f.IntVar(&cli.IntVar{
+		Name:    "pull-request-number",
+		Target:  &c.flagPullRequestNumber,
+		Example: "100",
+		Usage:   "The GitHub pull request number associated with this plan run.",
+	})
+
+	f.StringVar(&cli.StringVar{
+		Name:    "bucket-name",
+		Target:  &c.flagBucketName,
+		Example: "my-guardian-state-bucket",
+		Usage:   "The Google Cloud Storage bucket name to store Guardian plan files.",
+	})
+
+	f.BoolVar(&cli.BoolVar{
+		Name:    "protect-lockfile",
+		Target:  &c.flagProtectLockfile,
+		Default: true,
+		Example: "true",
+		Usage:   "Prevent modification of the Terraform lockfile.",
+	})
+
+	f.DurationVar(&cli.DurationVar{
+		Name:    "lock-timeout",
+		Target:  &c.flagLockTimeout,
+		Default: 10 * time.Minute,
+		Example: "10m",
+		Usage:   "The duration Terraform should wait to obtain a lock when running commands that modify state.",
+	})
+
+	return set
+}
+
+func (c *PlanRunCommand) Run(ctx context.Context, args []string) error {
+	logger := logging.FromContext(ctx)
+
+	f := c.Flags()
+	if err := f.Parse(args); err != nil {
+		return fmt.Errorf("failed to parse flags: %w", err)
+	}
+
+	if c.planFilename == "" {
+		c.planFilename = "tfplan.binary"
+	}
+
+	dir, err := normalizeWorkingDir(c.flagWorkingDirectory)
+	if err != nil {
+		return fmt.Errorf("failed to normalize working directory: %w", err)
+	}
+	c.flagWorkingDirectory = dir
+
+	args = f.Args()
+	if len(args) > 0 {
+		return fmt.Errorf("unexpected arguments: %q", args)
+	}
+
+	c.actions = githubactions.New(githubactions.WithWriter(c.Stdout()))
+	actionsCtx, err := c.actions.Context()
+	if err != nil {
+		return fmt.Errorf("failed to load github context: %w", err)
+	}
+
+	c.cfg = &Config{}
+	if err := c.cfg.MapGitHubContext(actionsCtx); err != nil {
+		return fmt.Errorf("failed to load github context: %w", err)
+	}
+	logger.Debugw("loaded configuration", "config", c.cfg)
+
+	c.githubClient = github.NewClient(
+		ctx,
+		c.GitHubFlags.FlagGitHubToken,
+		github.WithInitialRetryDelay(c.RetryFlags.FlagInitialRetryDelay),
+		github.WithMaxRetries(c.RetryFlags.FlagMaxRetries),
+		github.WithMaxRetryDelay(c.RetryFlags.FlagMaxRetryDelay),
+	)
+	c.terraformClient = terraform.NewTerraformClient(c.flagWorkingDirectory)
+
+	sc, err := storage.NewGoogleCloudStorage(
+		ctx,
+		storage.WithInitialRetryDelay(c.RetryFlags.FlagInitialRetryDelay),
+		storage.WithMaxRetryDelay(c.RetryFlags.FlagMaxRetryDelay),
+	)
+	if err != nil {
+		return fmt.Errorf("failed to create google cloud storage client: %w", err)
+	}
+	c.storageClient = sc
+
+	return c.Process(ctx)
+}
+
+func normalizeWorkingDir(dir string) (string, error) {
```

Comment:
Could you drop a comment on this function explaining what "normalize" means in this context?